packages/js/components/src/date-time-picker-control/date-time-picker-control.tsx

```diff
@@ -95,46 +85,42 @@ export const DateTimePickerControl: React.FC< DateTimePickerControlProps > = ( {
 		}
 
 		return default24HourDateTimeFormat;
-	} )();
+	}, [ dateTimeFormat, isDateOnlyPicker, is12HourPicker ] );
 
-	function parseMomentIso(
+	function parseAsISODateTime(
 		dateString?: string | null,
 		assumeLocalTime = false
 	): Moment {
-		if ( assumeLocalTime ) {
-			return moment( dateString, moment.ISO_8601, true ).utc();
-		}
-
-		return moment.utc( dateString, moment.ISO_8601, true );
+		return assumeLocalTime
+			? moment( dateString, moment.ISO_8601, true ).utc()
+			: moment.utc( dateString, moment.ISO_8601, true );
 	}
 
-	function parseMoment( dateString: string | null ): Moment {
+	function parseAsLocalDateTime( dateString: string | null ): Moment {
 		// parse input date string as local time;
 		// be lenient of user input and try to match any format Moment can
 		return moment( dateString );
 	}
 
-	function formatMomentIso( momentDate: Moment ): string {
-		return momentDate.utc().toISOString();
-	}
-
-	function formatMoment( momentDate: Moment ): string {
-		return formatDate( displayFormat, momentDate.local() );
-	}
-
-	function maybeForceTime( momentDate: Moment ): Moment {
-		if ( ! isDateOnlyPicker ) return momentDate;
+	const maybeForceTime = useCallback(
```

Comment:
Nice job wrapping this in a `useCallback`.  Naming is not bad here, would `maybeRoundTime` give a little more hint as to what this does?

Reply:
Good point... I will make a note of this and maybe include this in a future PR. Want to get this merged right away so I can pull the changes into https://github.com/woocommerce/woocommerce/pull/34538, so I can troubleshoot that with the latest and greatest `DateTimePickerControl`